File: src/main/java/com/ocbc/auctionservice/services/UserPrincipalDetailsService.java
package com.ocbc.auctionservice.services;

import com.ocbc.auctionservice.entities.User;
import com.ocbc.auctionservice.entities.UserPrincipal;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
public class UserPrincipalDetailsService implements UserDetailsService {

    @Autowired
    private UserService userService;

    @Override
    public UserDetails loadUserByUsername(String name) throws UsernameNotFoundException {
        User user = userService.getUser(name);
        return new UserPrincipal(user);
    }
}

File: src/main/kotlin/com/auction/service/authentication/jwt/UserPrincipalDetailsService.kt
package com.auction.service.authentication.jwt

import com.auction.service.services.UserService
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.stereotype.Component

@Component
class UserPrincipalDetailsService(
        private val userService: UserService
): UserDetailsService {

    override fun loadUserByUsername(username: String): UserDetails {
        val user = userService.getUser(username)
        return UserPrincipal(user)
    }

}